src/components/Card/Card.js

```diff
@@ -1,23 +1,57 @@
 import styled from 'styled-components/macro'
+import { useState } from 'react'
 import PropTypes from 'prop-types'
 
 Card.propTypes = {
   image: PropTypes.node.isRequired,
   title: PropTypes.string,
   text: PropTypes.string,
+  adress: PropTypes.string,
+  openingHours: PropTypes.string,
+  entry: PropTypes.string,
 }
 
-export default function Card({ text, title, image }) {
+export default function Card({
+  text,
+  title,
+  image,
+  adress,
+  openingHours,
+  entry,
+}) {
+  const [isExpanded, setIsExpanded] = useState(false)
+
   return (
-    <WrapperList>
-      <img src={image} alt="" />
-      <h2>{title}</h2>
-      <p>{text}</p>
-    </WrapperList>
+    <WrapperCardList isExpanded={isExpanded}>
+      <img isExpanded={isExpanded} src={image} alt="" />
+      <h2 isExpanded={isExpanded}>{title}</h2>
+      <p isExpanded={isExpanded}>{text}</p>
+
+      {isExpanded && (
+        <>
+          <ul>
+            <li>{adress}</li>
+            <li>{openingHours}</li>
+            <li>{entry}</li>
+          </ul>
+        </>
+      )}
+
+      <ButtonExpansion
+        isExpanded={isExpanded}
+        onClick={() => setIsExpanded(!isExpanded)}
+      >
+        {isExpanded ? 'Hide details' : 'Show details'}
```

Comment:
The isExpanded value as a child from the button (which renders hide details or show details) is taken from the state, not the button prop. unless you use isExpanded in the button component itself you don't need to set it as a prop here
```suggestion
      <ButtonExpansion
        onClick={() => setIsExpanded(!isExpanded)}
      >
        {isExpanded ? 'Hide details' : 'Show details'}
```